x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -108,144 +133,214 @@ protected void masterOperation(
                         if (indexerResp.isCreated()) {
-                            // 4.
+                            // 4. Make rollup index read-only and set the correct number of replicas
+                            final Settings settings = Settings.builder()
+                                .put(IndexMetadata.SETTING_BLOCKS_WRITE, true)
+                                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, sourceIndexMetadata.getNumberOfReplicas())
+                                .build();
+                            UpdateSettingsRequest updateSettingsReq = new UpdateSettingsRequest(settings, rollupIndexName);
+                            updateSettingsReq.setParentTask(parentTask);
                             client.admin().indices().updateSettings(updateSettingsReq, ActionListener.wrap(updateSettingsResponse -> {
                                 if (updateSettingsResponse.isAcknowledged()) {
-                                    // 5.
-                                    client.admin().indices().resizeIndex(resizeRequest, ActionListener.wrap(resizeResponse -> {
-                                        if (resizeResponse.isAcknowledged()) {
-                                            // 6.
-                                            publishMetadata(originalIndexName, tmpIndexName, rollupIndexName, listener);
+                                    // 5. Refresh rollup index
+                                    refreshIndex(rollupIndexName, parentTask, ActionListener.wrap(refreshIndexResponse -> {
+                                        if (refreshIndexResponse.getFailedShards() == 0) {
+                                            // 6. Add rollup index to data stream and publish rollup metadata
+                                            updateRollupMetadata(sourceIndexName, rollupIndexName, request, ActionListener.wrap(resp -> {
+                                                if (resp.isAcknowledged()) {
+                                                    // 9. Force-merge the rollup index to a single segment
+                                                    forceMergeIndex(
+                                                        rollupIndexName,
+                                                        parentTask,
+                                                        ActionListener.wrap(
+                                                            mergeIndexResp -> listener.onResponse(AcknowledgedResponse.TRUE),
+                                                            e -> listener.onFailure(
+                                                                new ElasticsearchException(
+                                                                    "Failed to force-merge index [" + rollupIndexName + "]",
+                                                                    e
+                                                                )
```

Comment:
Will this fail the request but not cause the destination index to be deleted? What will ILM think of that?

Reply:
In [393d7c2](https://github.com/elastic/elasticsearch/pull/85708/commits/393d7c2cd2c8c75f88970ca6caee55c2be9b1f08), I changed it so that it does not fail the rollup action.